app/services/id_token_builder.rb

```diff
@@ -55,24 +56,37 @@ def timestamp_claims
   end
 
   def acr
-    ial = identity.ial
-    case ial
-    when Idp::Constants::IAL_MAX then determine_ial_max_acr
-    when Idp::Constants::IAL1 then Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF
-    when Idp::Constants::IAL2 then Saml::Idp::Constants::IAL2_AUTHN_CONTEXT_CLASSREF
-    else
-      raise "Unknown ial #{ial}"
-    end
+    return nil unless identity.acr_values.present?
```

Comment:
Actually we could probably remove all the IAL computation in my previous comment if we dressed up this method like so:

```ruby
def acr
  if resolved_authn_context_result.ialmax?
    determine_ial_max_acr.name
  elsif resolved_authn_context_result.identity_proofing?
    Vot::LegacyComponentValues::IAL2.name
  else
    Vot::LegacyComponentValues::IAL1.name
  end
end
```

Reply:
And since this didn't stack up like I expected, this is the "previous comment" I am referring to: https://github.com/18F/identity-idp/pull/10142#discussion_r1503071957

Reply:
This way we:

1. Return `ial1` or `ial2` correctly in the IALMAX case
2. Return `ial1` or `ial2` even if the SP requests `loa1` or `loa3` or if the SP makes no `ial` related request
3. Only ever return one value in the ACR
4. Correctly return the `ial1` or `ial2` ACR values even if we introduce new ACR values or change names